app/notification/models/notification.py

```diff
@@ -18,10 +18,15 @@ class Notification:
     @staticmethod
     def process_notification_data(json_data):
         data = get_data(json_data)
-        notification_data = Notification(
-            template_type=data["type"],
-            contact_info=data["to"],
-            content=data["content"],
-        )
+        if (
+            "type" in data.keys()
+            and "to" in data.keys()
+            and "content" in data.keys()
+        ):
+            notification_data = Notification(
+                template_type=data.get("type"),
+                contact_info=data.get("to"),
+                content=data.get("content"),
+            )
 
-        return notification_data
+            return notification_data
```

Comment:
🔧 Should an exception or warning be raised if this condition isn't met? Otherwise how would the user know a malformed json has been sent.

Reply:
you could even swap this for a try-except block in routes.py and just have the error be passed up.

Reply:
also you could make the None return explicit - right now a Dev would have to understand that a function which doesn't explicitly return anything actually returns None/Void.

Reply:
Thanks, Jack, return None added to the class. 
It raises the warning in routes (400) with an example if data is not in the correct format.

Reply:
Just on this point—I think it would be good to talk about our logging in Python. I understand how a bad request is handled here Ram within the Route if theres a None return, but there is no obvious evidence in the logs that this has happened. It would be good to see logs for this kind of circumstance (maybe something like 'POST payload does not conform to notification schema')—what do you think @ramsharma-prog / @harryyo / @SentientHousePlant ?

Reply:
Now is returning an error.  Will try to add functionality to specify an error

Reply:
Added data & example data if no data or incorrect data. There is some in-built validation in the python-client library such as email validation etc.

Reply:
@Adam-W1 Logging has pretty good support in python using the [standard library](https://docs.python.org/3/library/logging.html#module-logging) . Let me know your thoughts - should be easy to add...